staging/src/k8s.io/sample-controller/README.md

```diff
@@ -126,6 +126,9 @@ type User struct {
 }
 ```
 
+Note, json tag `json:` is needed to generate informer externalversions package.
+Otherwise, informer internalversion package is generated.
```

Comment:
I think perhaps we want to be a little less specific here, while yes, the informer generation needs the JSON tags, there are many other generators that also need the tags. For example, applyconfiguration-gen, lister-gen and controller-gen also all treat a non json tagged member as internal, and therefore either create internal listers or omit the generated schemas.

Perhaps we want to go with something along the lines of 
```suggestion
Note, the JSON tag `json:` is required on all user facing fields within your type. Typically API types contain only user facing fields. When the JSON tag is omitted from the field, Kubernetes generators consider the field to be internal and will not expose the field in their generated external output. For example, this means that the field would not be included in a generated CRD schema.
```

On the informer `internalversions`/`externalversions` thing, it's my understanding that it only generates `internalversions` when it sees an internal field, and `externalversions` when it sees an external field, so I'm guessing when you ran into this, you had no json tags at all across your struct?

Reply:
Thanks! Have merged your suggestions!

> so I'm guessing when you ran into this, you had no json tags at all across your struct?

Yes, correct. I found json tag is needed when looking at source codes of code-generator.